x-pack/platform/plugins/shared/streams/server/lib/tasks/task_definitions/feature_identification.ts

```diff
@@ -0,0 +1,130 @@
+            } = await taskContext.getScopedClients({
+              request: runContext.fakeRequest,
+            });
+
+            // Get connector info for error enrichment
+            const connector = await inferenceClient.getConnectorById(connectorId);
+
+            try {
+              const [{ hits }, stream] = await Promise.all([
+                featureClient.getFeatures(name),
+                streamsClient.getStream(name),
+              ]);
+
+              const boundInferenceClient = inferenceClient.bindTo({ connectorId });
+              const esClient = scopedClusterClient.asCurrentUser;
+              const featureRegistry = getDefaultFeatureRegistry();
+
+              const promptsConfigService = new PromptsConfigService({
+                soClient,
+                logger: taskContext.logger,
+              });
+
+              const { featurePromptOverride, descriptionPromptOverride } =
+                await promptsConfigService.getPrompt();
+
+              const abortController = new AbortController();
+              abortControllers.set(runContext.taskInstance.id, abortController);
+
+              const results = await featureRegistry.identifyFeatures({
+                start,
+                end,
+                esClient,
+                inferenceClient: boundInferenceClient,
+                logger: taskContext.logger.get('feature_identification'),
+                stream,
+                features: hits,
+                signal: abortController.signal,
+                featurePromptOverride,
+                descriptionPromptOverride,
+              });
+
+              // I think I have to send the telemetry here
+
+              await taskClient.update<FeatureIdentificationTaskParams>({
+                ..._task,
+                status: 'completed',
+                task: {
+                  params: {
+                    connectorId,
+                    start,
+                    end,
+                  },
+                  payload: results,
+                },
+              });
+            } catch (error) {
+              const errorMessage = isInferenceProviderError(error)
+                ? formatInferenceProviderError(error, connector)
+                : error.message;
+
+              await taskClient.update<FeatureIdentificationTaskParams>({
+                ..._task,
+                status: 'failed',
+                task: {
+                  params: {
+                    connectorId,
+                    start,
+                    end,
+                  },
+                  error: errorMessage,
+                },
+              });
+            }
+          },
+          cancel: async () => {
+            if (abortControllers.has(runContext.taskInstance.id)) {
+              const abortController = abortControllers.get(runContext.taskInstance.id)!;
+              abortController.abort();
+              abortControllers.delete(runContext.taskInstance.id);
+            }
```

Comment:
`runContext` already has an AbortController, it is a new instance per task run.
And when `task.cancel()` is called `abortController.abort()` is called automatically.
Would it work here?